dvc/command/ls/__init__.py

```diff
@@ -37,7 +38,12 @@ def run(self):
             if self.args.show_json:
                 import json
 
-                ui.write(json.dumps(entries))
+                if sys.stdout.isatty():
+                    from rich.json import JSON
+
+                    ui.write(JSON.from_data(entries), styled=True)
+                else:
+                    ui.write(json.dumps(entries))
```

Comment:
Let's move this to [`dvc/ui/__init__.py`](https://github.com/iterative/dvc/blob/e81c88e501b79d2e1873b109f56332a4ad3903b7/dvc/ui/__init__.py) as `ui.write_json`, so that we can reuse it in other parts of dvc.

Also, support for `indent=` and `prettify=True|False` that keeps the old behaviour would be nice to have. 🙂

Reply:
Also, we may need to bump `rich` requirement, as `from_data` is only available since 10.9.0.

Reply:
Sure, that makes sense.  patch updated.